src/Encoding/FilteredStream.php

```diff
@@ -57,10 +57,22 @@
      */
     public function __construct(StreamInterface $stream, $readFilterOptions = null, $writeFilterOptions = null)
     {
+        $triggerDeprecationNotice = null !== $writeFilterOptions;
+
+        if (null === $readFilterOptions && in_array($this->readFilter(), ['convert.base64-encode', 'convert.base64-decode', 'convert.quoted-printable-encode', 'zlib.deflate', 'zlib.inflate'/*, 'bzip2.compress', 'bzip2.decompress'*/], true)) {
```

Comment:
Instead of listing functions like this, we could do:

```php
if (empty($readFilterOptions)) {
  $this->readFilterCallback = Filter\fun($this->readFilter());
} else {  
  $this->readFilterCallback = Filter\fun($this->readFilter(), $readFilterOptions);
}
```